core/base/src/TProcessID.cxx

```diff
@@ -414,7 +414,7 @@ void TProcessID::RecursiveRemove(TObject *obj)
    UInt_t uid = obj->GetUniqueID() & 0xffffff;
    if (obj == GetObjectWithID(uid)) {
       R__WRITE_LOCKGUARD(ROOT::gCoreMutex);
-      if (fgObjPIDs) {
+      if (fgObjPIDs && ((obj->GetUniqueID()&0xff000000)==0xff000000)) {
```

Comment:
I'm not very familiar with that code but it might make sense to add as a comment the PR description message.